photutils/psf/tests/test_photometry.py

```diff
@@ -497,3 +498,28 @@ def test_iterative_psf_photometry_inputs():
     with pytest.raises(ValueError, match=match):
         _ = IterativePSFPhotometry(psf_model, fit_shape, finder=finder,
                                    aperture_radius=4, maxiters=3.14)
+
+
+@pytest.mark.skipif(not HAS_SCIPY, reason='scipy is required')
+def test_out_of_bounds_centroids():
+    sources = Table()
+    sources['id'] = np.arange(4) + 1
+    sources['flux'] = 1
+    sources['x_0'] = [-1.4, 34.5, 14.2, -0.7]
+    sources['y_0'] = [13, -0.2, -1.6, 40]
+    sources['sigma'] = 3.1
+
+    shape = (51, 51)
+    data = make_gaussian_prf_sources_image(shape, sources)
+
+    psf_model = IntegratedGaussianPRF(flux=1, sigma=3.1)
+    fit_shape = (11, 11)
+    psfphot = PSFPhotometry(psf_model, fit_shape,
+                            aperture_radius=10)
+
+    phot = psfphot(data, init_params=Table({'x_init': sources['x_0'], 'y_init': sources['y_0']}))
```

Comment:
Also, you can now simply input `init_params=sources` as of https://github.com/astropy/photutils/pull/1566